api/all/src/main/java/io/opentelemetry/api/internal/OtelEncodingUtils.java

```diff
@@ -17,6 +17,7 @@
   private static final int NUM_ASCII_CHARACTERS = 128;
   private static final char[] ENCODING = buildEncodingArray();
   private static final byte[] DECODING = buildDecodingArray();
+  private static final boolean[] VALID_HEX = buildValidHexArray();
```

Comment:
64K array, not too big. JVM seems to be smart enough to not do any bounds check at all when indexing by `char` into an array of size `Character.MAX_VALUE`